### PR TITLE
Changed the type of root in data modules

### PR DESCRIPTION
**Pull request by samet-akcay**
# Description
- Change the type of `root` variable from `str` to `str | Path`.

# Motivation
- With this change, it will be easier to pass `Path` objects when using datamodules via the API.

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

### CODE REVIEW

**Review by djdameln — COMMENTED** [review]
Thanks, some minor comments

File: anomalib/data/avenue.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ class Avenue(AnomalibVideoDataModule):
 
     def __init__(
         self,
-        root: str,
+        root: Path | str,
         gt_dir: str,
```
[the remark this answers is not in the document]
Comment (samet-akcay) [reply anomalib/data/avenue.py:198]: Good catch!

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks

File: anomalib/data/avenue.py [hunk 2]
```diff
@@ -220,7 +220,7 @@ def __init__(
             seed=seed,
         )
 
-        self.root = Path(root)
+        self.root = Path(root) if isinstance(root, str) else root
```
[the remark this answers is not in the document]
Comment (samet-akcay) [reply anomalib/data/avenue.py:223]: Not sure either. Technically the old format is not wrong and mypy is happy with it. As you said, maybe we could stick to the existing approach

[the remark this answers is not in the document]
Comment (samet-akcay) [reply anomalib/data/avenue.py]: reverted back to the existing version.

**Review by djdameln — APPROVED** [review]
Thanks!

**Pull request merged**